### DISCUSSION TITLE
Google for authentication with self-hosted application?

### DISCUSSION THREAD

**Discussion by enisbu** [decaporg/decap-cms#7234] [Q&A] [answered]
Hey everyone, Is it possible to use an external provider like Google for authentication if I'm self-hosting my application and not using Netlify Identity?

(additional: Is it possible to use custom username & password input fields, instead of an external auth provider while also not using netlify as my hosting)

**Comment by martinjagodic** [comment, marked as answer]
This is theoretically possible, you need an external OAuth client. On [this list](https://decapcms.org/docs/external-oauth-clients/) you can find community-maintained projects. hopefully, some of those work for you. If not, you will have to create your own.
